### ISSUE TITLE
convert "createTEI" job to "upsertTEI" job using latest dhis2 adaptor version

### ISSUE THREAD

**Issue by aleksa-krolls** [OpenFn/logical-outcomes#10]
## Background
We have this old job that currently uses `createTEI(...)` operation: https://github.com/OpenFn/logical-outcomes/blob/master/care/createTEI_care_reach_impact_2024.js

For more project background, pls check out [this notes doc](https://docs.google.com/document/d/1UHTuyIPw82J2-6tiOeNizf9srPsO6E5Jenk1S5wf5bM/edit#heading=h.4wgbfqwbhuqc). 

## Request 
We want to **upsert** data now using the uid `SgQW3vpnhuL` --> please update the job & mappings accordingly. 
https://github.com/OpenFn/logical-outcomes/blob/271d967aa2b0dc6ca7953d46417e90dd19a9399f/care/createTEI_care_reach_impact_2024.js#L29-L32

FYI see here for other example DHIS2 jobs that upsert TEIs: 
- [See here](https://github.com/OpenFn/logical-outcomes/blob/master/care/upsertCareTEI.js) for another logical example job that uses the `upsert` operation... but note that this example is probably using an _old_ version of DHIS2 api + adaptor
- I think [this is a newer version](https://github.com/OpenFn/madagascar-dhis2-mtomady/blob/main/jobs/load-teis.js#L61) that uses a later version of DHIS2... and also looks up existing TEIs by searching for a specific attribute (in our case we'd want to see if TEIs already exist with matching values for the attribute with id `SgQW3vpnhu`)... and then create/update accordingly

## expression
You can overwrite the existing expression on `master` with a new commit. (The `master` branch is linked to [Georgi's Project](https://v1.openfn.org/projects/georgi-s-project/jobs/z4fMG4) in v1 platform for testing.)
https://github.com/OpenFn/logical-outcomes/blob/master/care/createTEI_care_reach_impact_2024.js

### adaptor
try to implement using latest dhis2 adaptor

### state
- config:`LO DHIS2 CARE PIIRS - TEST instance` (we'll get the UN & password from the client, but go ahead and start pls)
- data: use this message https://v1.openfn.org/projects/georgi-s-project/messages/065fd3b0-0ac9-7700-b105-4a424910c4d9

### To Test
Run updated job against this message --> it should successfully upsert the 2 TEI records and NOT create new records. Use the Search to look up the TEIs and make sure only 1 row is there - see screenshot to search within `REACH AND IMPACT FORM`. 
1. TEI row 1 has PIIRS UID `003763AFG` --> make sure TEI is not duplicated 
2. TEI row 2 has PIIRS UID `003763AFG` --> make sure TEI is not duplicated

![image](https://github.com/OpenFn/logical-outcomes/assets/16758106/1ae3f0f1-6605-466f-9b6d-6762554cae4c)

### Toggl
`LO DHIS2 Data Import 2024`

**Comment by aleksa-krolls** [comment]
@daissatou2 @mtuchi fyi testing notes are updated. Mtuchi, we'll let you know when we get the credential from the client.

**Comment by mtuchi** [comment]
I have pushed latest change on #11 for bulk upserting tracked entities. 
I am currently blocked with the following an error report
```
"errorReports": [
      {
        "message": "Non-unique attribute value `REDACTED` for attribute `qoDbMxYz38O`",
        "errorCode": "E1064",
        "trackerType": "TRACKED_ENTITY",
        "uid": "CW37v0BVQ5v"
      },
      {
        "message": "Non-unique attribute value `REDACTED` for attribute `SgQW3vpnhuL`",
        "errorCode": "E1064",
        "trackerType": "TRACKED_ENTITY",
        "uid": "xDllAw9mqD3"
      },
      {
        "message": "Non-unique attribute value `REDACTED` for attribute `qoDbMxYz38O`",
        "errorCode": "E1064",
        "trackerType": "TRACKED_ENTITY",
        "uid": "xDllAw9mqD3"
      },
      {
        "message": "Non-unique attribute value `003763AFG` for attribute `SgQW3vpnhuL`",
        "errorCode": "E1064",
        "trackerType": "TRACKED_ENTITY",
        "uid": "CW37v0BVQ5v"
      }
    ],
 ```
 I have a 30 min call on Monday with @taylordowns2000 to help me debug this error 
 Cc @daissatou2 @aleksa-krolls

**Comment by mtuchi** [comment]
The `upsert()` is working now, See #11 PR. But it's not a bulk upsert `(I couldn't get that to work 😞)`. The way it works now it will `upsert()` one record at a time using `each()` function.
cc @aleksa-krolls @gchakarov

**Comment by mtuchi** [comment]
I have merged the changes on #11, But i will keep this open until we have feedback from @gchakarov on `Non-unique attribute value '3763' for attribute vFRzB344yiC` cc @daissatou2 @aleksa-krolls
